STYLE.md

```diff
@@ -0,0 +1,364 @@
+**build_style** should come at the end of the variables section.
+
+The same applies for variables that take structured values like **conf_files**.
+There is no specific ordering for such variables. but it is preferred that
+multi-line versions of **conf_files** and other variables that are prone to
+requiring multiple lines like **make_dirs** should be at the end, with a blank
+space separating them from the other variables.
+
+When using **nopie**, **nocross** and **broken** make the value be something
+descriptive, if no other description is available, a build log showcasing the
+failure is acceptable.
+
+```
+broken="Segfaults at runtime"
+nocross="Tries to execute binary built for target on host"
+```
+
+If **alternatives** is used it should be the last line, if there are multiple
+alternatives presented then they should be 1 on each line, prefixed with
+whitespace.
+
+```
+checksum=blablablablablablablablablabla
+
+alternatives="foo:foo:usr/bin/foo
+ foo:bar:usr/bin/bar"
+```
+
+Variables that are not used by _xbps-src_ need to be prefixed with
+an underscore.
+
+## QUOTING
+
+Some rules of thumb to follow when quoting variables:
+
+- Don't quote variables that cannot ontain spaces. i.e: **pkgname**
+- Quote when there is a strong chance the variable contain spaces
+- Quote when a subshell is being run or a variable is being expanded
+
+> When expanding variables, put it in brackets.
+
+**pkgname** and **version** must **never** be quoted.
+
+Here is a list of variables that should **never** be quoted:
+
+- pkgname
+- version
+- revision
+- build_style
+- configure_script
+- make_cmd
+- make_build_target
+- make_check_target
+- make_install_target
+- disable_parallel_build
+- keep_libtool_archives
+- nodebug
+- repository
+- nostrip
+- noshlibprovides
+- noverifyrdeps
+- restricted
+- nopie
+- bootstrap
+- create_wrksrc
+
+Here is a list of variables that should be quoted in a specific condition:
+
+- archs
+	- If the value isn't **noarch**
+- wrksrc
+	- If the value doesn't contain spaces or variable expansion
+- build_wrksrc
+	- If the value doesn't contain spaces or variable expansion
+- checksum
+	- If there is more than 1 value
+
+Here is a list of variables that should always be quoted:
+
+- reverts
+- build_helper
+- configure_args
+- make_build_args
+- make_check_args
+- make_install_args
+- patch_args
+- hostmakedepends
+- makedepends
+- depends
+- checkdepends
+- short_desc
+- maintainer
+- license
+- homepage
+- distfiles
+- skip_extraction
+- conf_files
+- mutable_files
+- make_dirs
+- nostrip_files
+- skiprdeps
+- nocross
+- subpackages
+- broken
+- shlib_provides
+- shlib_requires
+- alternatives
+- font_dirs
+- dkms_modules
+- register_shell
+- tags
+- perl_configure_dirs
+- preserve
+- fetch_cmd
+- conflicts
+
+## CONTROL STRUCTURES
+
+Control structures are conditionals provided by the shell to
+control the flow of execution of the template. Examples:
+
+```
```

Comment:
first two example are valid only valid in the global scope, last two are only valid inside a phase. I think this can create confusion.

Reply:
Sadly i want to use real-world examples (all of these are taken from templates in srcpkgs/)